test/encoder/encoder_native_test.exs

```diff
@@ -1,14 +1,29 @@
 defmodule Encoder.NativeTest do
   use ExUnit.Case, async: true
+
+  import Membrane.Time
+
+  alias Membrane.H264.FFmpeg.Common
   alias Membrane.H264.FFmpeg.Encoder.Native, as: Enc
 
   test "Encode 1 240p frame" do
     in_path = "../fixtures/reference-100-240p.raw" |> Path.expand(__DIR__)
 
     assert {:ok, file} = File.read(in_path)
-    assert {:ok, ref} = Enc.create(320, 240, :I420, :fast, :high, -1, -1, 30, 1, 23)
+    assert {:ok, ref} = Enc.create(320, 240, :I420, :fast, :high, -1, -1, 1, 1, 23)
     assert <<frame::bytes-size(115_200), _tail::binary>> = file
-    assert {:ok, _dts_list, _frames} = Enc.encode(frame, 0, false, ref)
-    assert {:ok, [_dts], [_frame]} = Enc.flush(false, ref)
+
+    Enum.each(
+      0..4,
+      &Enc.encode(frame, Common.to_h264_time_base_truncated(seconds(&1)), false, ref)
+    )
+
+    assert {:ok, _dts_list, _pts, _frames} =
+             Enc.encode(frame, Common.to_h264_time_base_truncated(seconds(5)), false, ref)
```

Comment:
This separation doesn't make much sense.
1. You should assert on every encode call, it may return an error
2. The 5th call isn't any different
3. In all cases you assume no frames are returned from any call before flush

```suggestion
    for i <- 0..5 do
      pts = Common.to_h264_time_base_truncated(seconds(i))
      assert Enc.encode(frame, pts, false, ref) == {:ok, [], [], []}
    end
```